File: src/frontend/src/app/components/financeiro/conta-bancaria/model/conta-bancaria-dto.ts
export class ContaBancariaDTO {
  id?: string;
  nome: string;
  banco?: string;
  agencia?: string;
  numeroConta?: string;
  tipo: string; // CORRENTE, POUPANCA, CAIXA, INVESTIMENTO
  saldoInicial?: number;
  unidadeNegocioId?: string;
  unidadeNegocioNome?: string;
  unidadeNegocioCodigo?: string;
  ativa?: boolean;
  createdAt?: Date;
  updatedAt?: Date;
  createdBy?: string;
  updatedBy?: string;

  constructor(
    id?: string,
    nome?: string,
    banco?: string,
    agencia?: string,
    numeroConta?: string,
    tipo?: string,
    saldoInicial?: number,
    unidadeNegocioId?: string,
    unidadeNegocioNome?: string,
    unidadeNegocioCodigo?: string,
    ativa?: boolean,
    createdAt?: Date,
    updatedAt?: Date,
    createdBy?: string,
    updatedBy?: string
  ) {
    this.id = id;
    this.nome = nome ?? '';
    this.banco = banco;
    this.agencia = agencia;
    this.numeroConta = numeroConta;
    this.tipo = tipo ?? '';
    this.saldoInicial = saldoInicial;
    this.unidadeNegocioId = unidadeNegocioId;
    this.unidadeNegocioNome = unidadeNegocioNome;
    this.unidadeNegocioCodigo = unidadeNegocioCodigo;
    this.ativa = ativa;
    this.createdAt = createdAt;
    this.updatedAt = updatedAt;
    this.createdBy = createdBy;
    this.updatedBy = updatedBy;
  }
 
}

File: src/backend/src/main/java/br/com/grupopipa/gestaointegrada/financeiro/contabancaria/ContaBancariaDTO.java
package br.com.grupopipa.gestaointegrada.financeiro.contabancaria;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

import br.com.grupopipa.gestaointegrada.core.dto.DTO;

import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Builder
public class ContaBancariaDTO implements DTO {

  private UUID id;
  private String nome;
  private String banco;
  private String agencia;
  private String numeroConta;
  private String tipo; // CORRENTE, POUPANCA, CAIXA, INVESTIMENTO
  private BigDecimal saldoInicial;
  private UUID unidadeNegocioId;
  private Boolean ativa;
  private LocalDateTime createdAt;
  private LocalDateTime updatedAt;
  private String createdBy;
  private String updatedBy;
}
